### PR TITLE
update demo to match the tutorial

### PR DESCRIPTION
**Pull request by notwaldorf**
Bam. :fire: :tada: 

It uses a tiny helper element that uses `marked-element` and `prism-highlighter` to spit the code out, and render it.

It looks like this (and so-on for all the examples in the tutorial at https://elements.polymer-project.org/guides/flex-layout):

screen shot 2015-10-29 at 2 59 48 pm

### CODE REVIEW

**Comment by notwaldorf** [comment]
Hmm, hold off on this PR. Moving the `Polymer.domInnerHTML.getInnerHTML` bit to `attached` not `ready` actually gets the content after it's gone through the styling system, so it ends up looking like this, which is bad:
screen shot 2015-10-30 at 1 02 22 pm

**Comment by cdata** [comment]
º\\(:sparkles:)_(:sparkles:)/º

File: demo/demo-snippet.html [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          type: String,
+          value: ''
+        }
+      },
+
+      attached: function() {
+        var template = Polymer.dom(this).queryDistributedElements('template')[0];
+
+        var snippet = Polymer.domInnerHTML.getInnerHTML(template);
+        this._markdown = '```\n' + snippet + '\n' + '```';
```
Comment (cdata) [inline demo/demo-snippet.html]: 50 1337! :computer:

File: demo/index.html [hunk 2]
```diff
@@ -12,31 +12,378 @@
...
+          }
+        </style>
+
+        <div class="container" id="demo3">
+          <div>one</div>
+          <div class="flexchild">two (flex)</div>
+          <div>three</div>
+        </div>
+      </template>
+		</demo-snippet>
```
Comment (cdata) [inline demo/index.html]: I find your lack of
   :full_moon:     correct indentation
disturbing.

File: demo/demo-snippet.html [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        _markdown: {
+          type: String,
+          value: ''
+        }
+      },
+
+      attached: function() {
+        var template = Polymer.dom(this).queryDistributedElements('template')[0];
+
+        var snippet = Polymer.domInnerHTML.getInnerHTML(template);
```
Comment (cdata) [inline demo/demo-snippet.html]: Is this dark magic documented anywhere? If not, it totally should be...

**Comment by cdata** [comment]
I love it. LGTM :+1:

File: demo/demo-snippet.html (hunk 3 above)
Comment (notwaldorf) [inline demo/demo-snippet.html]: Good question. I don't see it anywhere, so I'm going to pass this hot potato to @arthurevans  :)

Comment (arthurevans) [inline demo/demo-snippet.html]: What the tarnation is `domInnerHTML`? Is that even a public API?
Is there some reason that shouldn't be Polymer.dom(template).innerHTML?

Comment (notwaldorf) [inline demo/demo-snippet.html]: I think you're right. Initially I wasn't using a `<template>`, I was just passing the content in as distributed children, in which case `domInnerHTML` does the right thing. I guess it's probably best that I update this to use the simpler `Polymer.dom(template).innerHTML` now

**Pull request merged**